### PR TITLE
ENH: Make crossval html report generic to all hyperdrive runs

### PR DESCRIPTION
**Pull request by kenza-bouzid**
* Add commandline arguments to parametrise child runs arg identifier and primary metric to get best_epoch.
* Rename all crossval functions to hyperdrive ones to make it generic

### CODE REVIEW

File: hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py [hunk 1]
```diff
@@ -62,7 +69,8 @@ def generate_html_report(parent_run_id: str, output_dir: Path,
     if num_classes > 1:
         base_metrics_list = [MetricsKey.ACC, MetricsKey.AUROC]
```
Comment (harshita-s) [inline hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py:70]: the lists for metrics need to be updated for both binary and multi-class cases

File: hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py [hunk 2]
```diff
@@ -242,6 +252,9 @@ def render_confusion_matrices(report: HTMLReport, heading: str, level: int, clas
                                                                     "in the generated report.")
     parser.add_argument('--overwrite', action='store_true', help="Forces (re)download of metrics and output files, "
                                                                  "even if they already exist locally.")
+    parser.add_argument("--hyper_arg_name", default="crossval_index",
+                        help="Name of the Hyperdrive argument used for indexing the child runs.")
+    parser.add_argument("--primary_metric", default="auroc", help="Name of the reference metric to optimise.")
```
Comment (harshita-s) [inline hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py:257]: can we use `MetricsKey.AUROC` as default?

File: hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py [hunk 3]
```diff
@@ -190,35 +190,35 @@ def _plot_crossval_roc_and_pr_curves(crossval_dfs: Dict[int, pd.DataFrame], roc_
         # assert len(non_unique_slides) == 0
         scores = tile_scores.first()
 
-        plot_roc_curve(labels, scores, legend_label=f"Fold {k}", ax=roc_ax)
-        plot_pr_curve(labels, scores, legend_label=f"Fold {k}", ax=pr_ax)
+        plot_roc_curve(labels, scores, legend_label=f"Child {k}", ax=roc_ax)
+        plot_pr_curve(labels, scores, legend_label=f"Child {k}", ax=pr_ax)
     legend_kwargs = dict(edgecolor='none', fontsize='small')
```
Comment (harshita-s) [inline hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py:195]: not sure why codecov complains here. Is this not included in tests?

File: hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py [hunk 4]
```diff
@@ -190,35 +190,35 @@ def _plot_crossval_roc_and_pr_curves(crossval_dfs: Dict[int, pd.DataFrame], roc_
...
     format_pr_or_roc_axes('roc', roc_ax)
     format_pr_or_roc_axes('pr', pr_ax)
 
 
-def plot_crossval_roc_and_pr_curves(crossval_dfs: Dict[int, pd.DataFrame],
-                                    scores_column: str = ResultsKey.PROB) -> Figure:
-    """Plot ROC and precision-recall curves for multiple cross-validation runs.
+def plot_hyperdrive_roc_and_pr_curves(hyperdrive_dfs: Dict[int, pd.DataFrame],
+                                      scores_column: str = ResultsKey.PROB) -> Figure:
+    """Plot ROC and precision-recall curves for multiple hyperdrive cgild runs.
```
Comment (harshita-s) [inline hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py:205]: typo in `child`

File: hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py (hunk 1 above)
Comment (kenza-bouzid) [reply hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py:70]: Good point, done!

File: hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py (hunk 2 above)
Comment (kenza-bouzid) [reply hi-ml-cpath/src/health_cpath/scripts/generate_crossval_report.py:257]: Done!

File: hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py (hunk 3 above)
Comment (kenza-bouzid) [reply hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py:195]: It doesn't appear that we have tests for these functions. Might be out of scope of this PR?

Comment (harshita-s) [reply hi-ml-cpath/src/health_cpath/utils/analysis_plot_utils.py:195]: ok, resolving this now. Perhaps open a ticket for the tests?

**Review by harshita-s — APPROVED** [review]

**Pull request merged**